ibis/pyspark/tests/test_array.py

```diff
@@ -0,0 +1,231 @@
+        ),
+    ],
+)
+def test_array_slice_scalar(client, start, stop):
+    raw_value = [-11, 42, 10]
+    value = ibis.literal(raw_value)
+    expr = value[start:stop]
+    result = client.execute(expr)
+    expected = raw_value[start:stop]
+    assert result == expected
+
+
+@pytest.mark.parametrize('index', [1, 3, 4, 11, -11])
+def test_array_index(client, index):
+    table = client.table('array_table')
+    expr = table[table.array_int[index].name('indexed')]
+    result = expr.execute()
+
+    df = table.compile().toPandas()
+    expected = pd.DataFrame(
+        {
+            'indexed': df.array_int.apply(
+                lambda x: x[index] if -len(x) <= index < len(x) else None
+            )
+        }
+    )
+    tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize('index', [1, 3, 4, 11])
+def test_array_index_scalar(client, index):
+    raw_value = [-10, 1, 2, 42]
+    value = ibis.literal(raw_value)
+    expr = value[index]
+    result = client.execute(expr)
+    expected = raw_value[index] if index < len(raw_value) else None
+    assert result == expected
+
+
+@pytest.mark.parametrize('op', [lambda x, y: x + y, lambda x, y: y + x])
+def test_array_concat(client, op):
+    table = client.table('array_table')
+    x = table.array_int.cast('array<string>')
+    y = table.array_str
+    expr = op(x, y)
+    result = expr.execute()
+
+    df = table.compile().toPandas()
+    expected = op(
+        df.array_int.apply(lambda x: list(map(str, x))),
+        df.array_str,
+    ).rename('tmp')
+    tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize('op', [lambda x, y: x + y, lambda x, y: y + x])
+def test_array_concat_scalar(client, op):
+    raw_left = [1, 2, 3]
+    raw_right = [3, 4]
+    left = ibis.literal(raw_left)
+    right = ibis.literal(raw_right)
+    expr = op(left, right)
+    result = client.execute(expr)
+    assert result == op(raw_left, raw_right)
+
+
+@pytest.mark.parametrize('n', [1, 3, 4, 7, -2])  # negative returns empty list
+@pytest.mark.parametrize('mul', [lambda x, n: x * n, lambda x, n: n * x])
+def test_array_repeat(client, n, mul):
+    table = client.table('array_table')
+
+    expr = table.projection([mul(table.array_int, n).name('repeated')])
+    result = expr.execute()
+
+    df = table.compile().toPandas()
+    expected = pd.DataFrame({'repeated': df.array_int * n})
+    tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize('n', [1, 3, 4, 7, -2])  # negative returns empty list
+@pytest.mark.parametrize('mul', [lambda x, n: x * n, lambda x, n: n * x])
+def test_array_repeat_scalar(client, n, mul):
+    raw_array = [1, 2]
+    array = ibis.literal(raw_array)
+    expr = mul(array, n)
+    result = client.execute(expr)
+    expected = mul(raw_array, n)
+    assert result == expected
+
+
+def test_array_collect(client):
+    table = client.table('array_table')
+    expr = table.group_by(table.key).aggregate(
+        collected=table.array_int.collect()
```

Comment:
What does this do?

Reply:
This collects the arrays in `array_int` column after grouping by key. So for example, `k1` corresponds to `[1, 2, 3]` and `[]` so the result will be one array `[1, 2, 3]`.

Reply:
Does it just concat all the arrays? i.e. if `k1` corresponds `[1, 2 ,3]` and `[4, 5]` does it return `[1, 2, 3, 4 ,5]`?

Reply:
Sorry, I misspoke. The result is a collection of arrays (e.g. `[[1, 2, 3], [4, 5]]`). Updated the test data to also collect a non-empty array for the same key.